src/rendering/luaconsole.cpp

```diff
@@ -716,88 +496,106 @@ void LuaConsole::render() {
     );
 
     // Render the current command
-    std::string currentCommand = _commands.at(_activeCommand);
-    // We chop off the beginning and end of the string until it fits on the screen (with
-    // a margin) this should be replaced as soon as the mono-spaced fonts work properly.
-    // Right now, every third character is a bit wider than the others
+    std::string currentCommand = _commands[_activeCommand];
+
+    // Render the command and suggestions with line breaks where required, + 2 accounts
+    // for the '> ' characters in the beginning of the command
+    const size_t totalCommandSize = 2 + currentCommand.size() +
+        _autoCompleteState.suggestion.size();
+    // Scalefactor 0.925f chosen arbitraily to fit characters on screen with some margin
+    const size_t nCharactersPerRow = std::max(
+        static_cast<size_t>(1),
+        static_cast<size_t>(res.x * 0.925f / static_cast<float>(_font->glyph('m')->width))
+    );
+    size_t nCommandRows = static_cast<size_t>(
+        std::ceil(static_cast<double>(totalCommandSize) / nCharactersPerRow)
+    );
+    // We're intrested in the zero based index when computing the input position
+    // If the characters fit on one line we should not add any extra rows
+    nCommandRows = nCommandRows > 1 ? nCommandRows - 1 : 0;
+
+    // The command is split in 3 parts to render the suggestion in a different color
+    std::string firstPart = _autoCompleteState.insertPosition != 0 ?
+        currentCommand.substr(0, _autoCompleteState.insertPosition) : currentCommand;
+    std::string secondPart = _autoCompleteState.insertPosition != 0 ?
+        currentCommand.substr(_autoCompleteState.insertPosition) : "";
+
+    // We pad the strings with empty spaces so that each part is rendered in their correct
+    // positions, even if linebreaks are added
+    // Pad suggestion before and after with ' '
+    const std::string suggestion = std::string(firstPart.size() + 2, ' ') +
+        _autoCompleteState.suggestion + std::string(secondPart.size(), ' ');
+    // Pad first part at the end with ' '
+    firstPart.insert(firstPart.end(), totalCommandSize - firstPart.size(), ' ');
+    // Pad second part in the beginning with ' '
+    secondPart.insert(secondPart.begin(), totalCommandSize - secondPart.size(), ' ');
+
+    auto linebreakCommand = [nCharactersPerRow](const std::string& s) {
```

Comment:
This lambda could use an explanatory comment about what it does. It seems to be called for all rendered and is hence important enough to be explained :)